framework/test/driver/BundleGetSymbolTest.cpp

```diff
@@ -0,0 +1,119 @@
+/*=============================================================================
+
+  Library: CppMicroServices
+
+  Copyright (c) The CppMicroServices developers. See the COPYRIGHT
+  file at the top-level directory of this distribution and at
+  https://github.com/CppMicroServices/CppMicroServices/COPYRIGHT .
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+=============================================================================*/
+
+#include "cppmicroservices/Bundle.h"
+#include "cppmicroservices/BundleActivator.h"
+#include "cppmicroservices/BundleContext.h"
+#include "cppmicroservices/Framework.h"
+#include "cppmicroservices/FrameworkFactory.h"
+#include "cppmicroservices/GetBundleContext.h"
+
+#include "cppmicroservices/SharedLibrary.h"
+
+#include "TestingConfig.h"
+#include "TestingMacros.h"
+#include "TestUtils.h"
+
+#include <stdexcept>
+
+using namespace cppmicroservices;
+
+void TestGetSymbolInvalidBundleInput()
+{
+    // Test for invalid bundle
+    
+    Bundle b;
+    US_TEST_FOR_EXCEPTION(std::invalid_argument , b.GetSymbol(nullptr,""));
+}
+
+void TestGetSymbolInvalidInput()
+{
+    FrameworkFactory factory;
+    auto f = factory.NewFramework();
+    f.Start();
```

Comment:
Should RAII be used here to avoid the Start/Stop pattern?

Reply:
Every test has a different requirement. e.g. Invalid Bundle, Valid bundle which is not started/installed and valid bundle with lifetime. Give this, I think SetUp and TearDown might not work we expect do things that are superfluous for some tests. I am keeping the tests as it is for now. I will keep in mind to add SetUp and TearDown henceforth.